packages/warehouses/src/warehouseClients/AthenaWarehouseClient.ts

```diff
@@ -0,0 +1,320 @@
+                AND table_schema = lower('${schema}')
+                AND table_name = lower('${table}')
+            ORDER BY 1, 2, 3, ordinal_position`;
+
+const convertDataTypeToDimensionType = (
+    type: AthenaTypes | string,
+): DimensionType => {
+    const typeWithoutTimePrecision = type.replace(/\(\d\)/, '');
+    switch (typeWithoutTimePrecision) {
+        case AthenaTypes.BOOLEAN:
+            return DimensionType.BOOLEAN;
+        case AthenaTypes.TINYINT:
+            return DimensionType.NUMBER;
+        case AthenaTypes.SMALLINT:
+            return DimensionType.NUMBER;
+        case AthenaTypes.INTEGER:
+            return DimensionType.NUMBER;
+        case AthenaTypes.BIGINT:
+            return DimensionType.NUMBER;
+        case AthenaTypes.REAL:
+            return DimensionType.NUMBER;
+        case AthenaTypes.DOUBLE:
+            return DimensionType.NUMBER;
+        case AthenaTypes.DECIMAL:
+            return DimensionType.NUMBER;
+        case AthenaTypes.DATE:
+            return DimensionType.DATE;
+        case AthenaTypes.TIMESTAMP:
+            return DimensionType.TIMESTAMP;
+        case AthenaTypes.TIMESTAMP_TZ:
+            return DimensionType.TIMESTAMP;
+        default:
+            return DimensionType.STRING;
+    }
+};
+
+export class AthenaWarehouseClient extends WarehouseBaseClient<CreateAthenaCredentials> {
+    connectionOptions: Pick<
+        CreateAthenaCredentials,
+        'outputLocation' | 'workgroup' | 'extraBotoArgs' | 'database'
+    >;
+
+    client: AthenaClient;
+
+    constructor(credentials: CreateAthenaCredentials) {
+        super(credentials);
+        const credObj: { credentials?: any } = {};
+        if (credentials.awsAccessKeyId && credentials.awsSecretKey) {
+            credObj.credentials = {
+                accessKeyId: credentials.awsAccessKeyId,
+                secretAccessKey: credentials.awsSecretKey,
+            };
+        }
+
+        this.client = new AthenaClient({
+            region: credentials.awsRegion,
+            ...credObj,
+        });
+
+        this.connectionOptions = {
+            outputLocation: credentials.outputLocation,
+            workgroup: credentials.workgroup,
+            database: credentials.database,
+            extraBotoArgs: credentials.extraBotoArgs,
+        };
+    }
+
+    private async checkQueryExequtionStateAndGetData(
+        QueryExecutionId: string,
+    ): Promise<any> {
+        const command = new GetQueryExecutionCommand({ QueryExecutionId });
+        const response = await this.client.send(command);
+
+        const state = response.QueryExecution?.Status?.State;
+
+        if (
+            state === QueryExecutionState.QUEUED ||
+            state === QueryExecutionState.RUNNING
+        ) {
+            // In my case, queries run no faster than 800-900ms, which is why I set a 1000ms timeout
+            await new Promise<void>((res) => {
+                setTimeout(() => res(), 1000);
```

Comment:
Could we make the timeout configurable?
Also let me get it right, if a query takes more than 1 second will you timeout?

Reply:
Ah I got it, it's the waited time that determine how often we check when the query result is ready. In dbt-athena we call this **poll_interval**.

My suggestion would be to make the value configurable, with a default value of 0.5.
I've seen very optimized queries running below 1sec.
Also, as suggested by @magnew let's add a comment please, to give more context 🙏🏻